store/manager.go

```diff
@@ -147,12 +167,47 @@ func (m *Manager) Put(ctx context.Context, cm commitments.CommitmentMode, key, v
 
 // putEigenDAMode ... disperses blob to EigenDA backend
 func (m *Manager) putEigenDAMode(ctx context.Context, value []byte) ([]byte, error) {
-	if m.eigenda != nil {
-		m.log.Debug("Storing data to EigenDA backend")
+	if !m.writeV2 { // disperse v1
+		m.log.Info("Storing data to EigenDA backend")
 		return m.eigenda.Put(ctx, value)
 	}
 
-	return nil, errors.New("no DA storage backend found")
+	return m.eigendaV2.Put(ctx, value)
+}
+
+func (m *Manager) getEigenDAMode(ctx context.Context, v commitments.EigenDACommit, key []byte) ([]byte, error) {
+	switch v {
+	case commitments.CertV0:
+		m.log.Debug("Reading blob from EigenDAV1 backend")
+		data, err := m.eigenda.Get(ctx, key)
+		if err == nil {
+			// verify v1 (payload, cert)
+			err = m.eigenda.Verify(ctx, key, data)
+			if err != nil {
+				return nil, err
+			}
+			return data, nil
+		}
+
+		return nil, err
+
+	case commitments.CertV1:
+		m.log.Debug("Reading blob from EigenDAV2 backend")
+		data, err := m.eigendaV2.Get(ctx, key)
+		if err == nil {
+			// verify v2 (payload, cert)
+			err = m.eigendaV2.Verify(ctx, key, data)
```

Comment:
I think we ought to have a different ordering here (since cert verification will likely be faster than retrieval of the full blob, right?):

1. Verify the cert on chain
2. If the cert is verified, only then proceed to try to retrieve the blob, from the available backend sources in order of priority

Reply:
this makes sense but where do we verify the data commitments for the blob?

Reply:
The commitments are verified as part of the payload fetch process, [here](https://github.com/Layr-Labs/eigenda/blob/5eab8ddece7ea2104d1ce2e082e63ed4e4573f80/api/clients/v2/relay_payload_retriever.go#L149) for relay retrieval, and [here](https://github.com/Layr-Labs/eigenda/blob/5eab8ddece7ea2104d1ce2e082e63ed4e4573f80/api/clients/v2/validator_payload_retriever.go#L161) for validator node retrieval